BibBuddy/app/src/main/java/de/bibbuddy/ImportBibTex.java

```diff
@@ -0,0 +1,306 @@
+package de.bibbuddy;
+
+import android.content.Context;
+import android.net.Uri;
+import androidx.annotation.NonNull;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * The ImportBibTex is responsible for the import of a
+ * BibTex file.
+ * It contains methods for reading and parsing the
+ * contents of the imported file.
+ *
+ * @author Silvia Ivanova
+ */
+public class ImportBibTex {
+
+  private final String [] bibTags;
+  private final Context context;
+  private final HashMap<String, String> bibTagValue;
+
+  /**
+   * Constructor for the Import of BibTeX file.
+   *
+   * @param context      current context
+   * @param bibTags      Array of BibTeX tags as String (e.g. "author=", "title=")
+   * @param bibTagValue  a HashMap to store the parsed BibTeX data
+   *                     Key is BibTeX tag.
+   *                     Value is the parsed text content for the corresponding BibTex tag.
+   */
+  public ImportBibTex(Context context, String [] bibTags, HashMap<String, String> bibTagValue) {
+    this.context = context;
+    this.bibTags = bibTags;
+    this.bibTagValue = bibTagValue;
```

Comment:
You don't need to pass those tags as parameters from the BookFragment. You can declare them directly in the ImportBibTex class.
```suggestion
```